java/src/jmri/jmrit/ToolsMenu.java

```diff
@@ -217,14 +219,18 @@ public ToolsMenu() {
                     updateProgrammerStatus(evt);
                 });
         InstanceManager.getList(GlobalProgrammerManager.class).forEach(m -> m.addPropertyChangeListener(this::updateProgrammerStatus));
+
+        for (Plugin p : InstanceManager.getDefault(PluginManager.class).getPlugins()) {
+            p.addToolsMenu(this);
```

Comment:
This means that every `PlugIn` has to add to the Tools menu.  Does that make this mechanism less general?  Maybe a check as to whether the plugin object wants to add to the menu?

Reply:
The `jmri.Plugin` class has both its methods as default methods with empty bodies. So you don't need to override them. If the plugin doesn't want to add anything to the Tools menu, it can skip implementing the `addToolsMenu(Object)` method.

But maybe the wording of the Javadoc of the `addToolsMenu(Object)` method should explain this better?